File: Travel/UI/travel_admin/lib/model/route/route.dart
import 'dart:ffi';

import 'package:json_annotation/json_annotation.dart';
import 'package:travel_admin/model/agency/agency.dart';
import 'package:travel_admin/model/city/city.dart';
import 'package:travel_admin/model/enums/bus_type.dart';
part 'route.g.dart';

@JsonSerializable()
class RouteModel {
  int? id;
  int? numberOfSeats;
  String? adminId;
  double? childPrice;
  double? adultPrice;
  int? toCityId;
  int? fromCityId;
  CityModel? toCity;
  CityModel? fromCity;
  String? travelTime;
  String? departureTime;
  String? arrivalTime;
  int? busType;
  int? agencyId;
  AgencyModel? agency;
  DateTime? validFrom;
  DateTime? validTo;
  RouteModel(
      this.id,
      this.numberOfSeats,
      this.adminId,
      this.adultPrice,
      this.childPrice,
      this.toCity,
      this.toCityId,
      this.fromCity,
      this.fromCityId,
      this.travelTime,
      this.departureTime,
      this.arrivalTime,
      this.agency,
      this.agencyId,
      this.validFrom,
      this.validTo,
      this.busType);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory RouteModel.fromJson(Map<String, dynamic> json) =>
      _$RouteModelFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$RouteModelToJson(this);
}

File: Travel/Travel.Services/Database/Route.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Travel.Models.Enums;
using Travel.Services.Domain.Base;

namespace Travel.Services.Database
{
    public class Route : BaseSoftDeleteEntity
    {
        public long Id { get; set; }
        public int NumberOfSeats { get; set; }
        public int AvailableSeats { get; set; }
        public double AdultPrice { get; set; }
        public double ChildPrice { get; set; }
        public long FromCityId { get; set; }
        public City FromCity { get; set; }
        public long ToCityId { get; set; }
        public City ToCity { get; set; }
        public string TravelTime { get; set; }
        public TimeSpan DepartureTime { get; set; }
        public TimeSpan ArrivalTime { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime ValidTo { get; set; }
        public long AgencyId { get; set; }
        public Agency Agency { get; set; }
        public BusType BusType { get; set; }
        public virtual ICollection<RouteTicket> RouteTickets { get; set; } = new List<RouteTicket>();
        public virtual ICollection<SavedRoutes> SavedRoutes { get; set; } = new List<SavedRoutes>();
    }
}
